ERCS/erc-6224.md

```diff
@@ -13,241 +13,613 @@ requires: 1967, 5750
 
 ## Abstract
 
-The EIP standardizes the management of smart contracts within the decentralized application ecosystem. It enables protocols to become upgradeable and reduces their maintenance threshold. This EIP additionally introduces a smart contract dependency injection mechanism to audit dependency usage, to aid larger composite projects.
+The EIP introduces a system of smart contracts that a decentralized protocol may use to manage its on-chain dependencies. The system also provides a convenient way of upgrading registered smart contracts, together with specifying custom access control permissions required by a protocol's business logic.
```

Comment:
I'd like to see a bit more of a technical overview in the abstract. This section should give the reader a high-level understanding of how the proposed solution is implemented.

Reply:
Roger that.